implementation/elasticsearch/src/main/java/net/opentsdb/meta/NamespacedAggregatedDocumentQueryBuilder.java

```diff
@@ -214,13 +216,26 @@ FilterBuilder getAnyFieldFilter(final AnyFieldRegexFilter filter) {
   }
   
   AggregationBuilder<?> metricAgg(final QueryFilter filter) {
+    QueryFilter metric_only_filter = null;
+    if (filter instanceof ChainFilter) {
+      BoolFilterBuilder builder = FilterBuilders.boolFilter();
+      // everything has to be a should here.
+      for (final QueryFilter sub_filter : ((ChainFilter) filter).getFilters()) {
```

Comment:
This will only pick the last metric filter. What if we have a query like 
```
chain {
  metric1,
  metric2,
  chain {
     tags1,
     tags2
   }
}
```?